lib/beaker/host/mac/pkg.rb

```diff
@@ -10,6 +10,25 @@ def install_package(name, cmdline_args = '', version = nil)
     execute("installer -pkg /Volumes/#{name}/#{name}.pkg -target /")
   end
 
+  # Install a package from a specified dmg
+  #
+  # @param [String] dmg_file      The dmg file, including path if not
+  #                               relative. Can be a URL.
+  # @param [String] pkg_base      The base name of the directory that the dmg
+  #                               attaches to under `/Volumes`
+  # @param [String] pkg_name      The name of the package file that should be
+  #                               used by the installer
+  # @example: Install vagrant from URL
+  #   mymachost.generic_install_dmg('https://releases.hashicorp.com/vagrant/1.8.4/vagrant_1.8.4.dmg', 'Vagrant', 'Vagrant.pkg')
+  def generic_install_dmg(dmg_file, pkg_base, pkg_name)
```

Comment:
I see what you mean from a conceptual level, but I'm just talking about duplicate code in the mac implementation of both methods. If we replaced the current `install_package` (just above, L8) with this, would it not function exactly the same as before, except that you'd be able to use `install_package` with URLs as well?
```ruby
# current version:
def install_package(name, cmdline_args = '', version = nil)
  execute("hdiutil attach #{name}.dmg")
  execute("installer -pkg /Volumes/#{name}/#{name}.pkg -target /")
end

# refactored:
def install_package(name, cmdline_args = '', version = nil)
  generic_install_dmg("#{name}.dmg", name, "#{name}.pkg")
end
```
I'll defer to your knowledge here, especially from your recent testing for this change. I just wanted to bring that up as a way to potentially remove duplicate code. Let me know what you decide looking at this, and hopefully we'll get this merged & released soon.